learn/writing-secure-ballerina-code.md

```diff
@@ -76,60 +76,51 @@ Security-sensitive functions and remote methods of Ballerina standard libraries
-public remote function select(@untainted string sqlQuery, 
-                              typedesc<record{}>? recordType, 
-                              Param... parameters) 
-                        returns @tainted table<record {}>|Error
+public remote function query(@untainted string|sql:ParameterizedQuery sqlQuery, typedesc<record {}>? rowType = ())
+    returns @tainted stream <record {}, sql:Error>
 ```
 
 The following example constructs an SQL query with a tainted argument:
 
 ```ballerina
-import ballerinax/java.jdbc;
+import ballerina/java.jdbc;
+import ballerina/sql;
 
 type ResultStudent record {
     string name;
 };
 
 public function main(string... args) {
 
-    jdbc:Client testDB = new({
-        url: "jdbc:mysql://localhost:3306/testdb",
-        username: "test",
-        password: "test",
-        poolOptions: { maximumPoolSize: 5 },
-        dbOptions: { useSSL: false }
-    });
-
-   // Construct student ID based on user input.
-   string studentId = "S_" + args[0];
-
-   // Execute select query using the untrusted (tainted) student ID
-   var dt = testDB->select("SELECT NAME FROM STUDENT WHERE ID = " + studentId,
-                           ResultStudent);
-   testDB.stop();
+    jdbc:Client|sql:Error jdbcClient = new("jdbc:mysql://localhost:3306/testdb", "test", "test");
+    if (jdbcClient is jdbc:Client) {
+        // Construct student ID based on user input.
+        string studentId = "S_" + args[0];
+
+        // Execute select query using the untrusted (tainted) student ID
+        stream<record{}, error> resultStream = jdbcClient->query("SELECT NAME FROM STUDENT WHERE ID = " + studentId);
+        error? status = jdbcClient.close();
+    }
 }
 ```
 
 The Ballerina compiler will generate an error:
 
 ```
-tainted value passed to sensitive parameter 'sqlQuery'
+tainted value passed to untainted parameter 'sqlQuery'
 ```
 
 In order to compile, the program is modified to use query parameters:
 
 ```ballerina
-jdbc:Parameter paramId = {sqlType:jdbc:TYPE_VARCHAR, value:studentId};
-var dt = testDB->select("SELECT NAME FROM STUDENT WHERE ID = ?", ResultStudent,
-                        paramId);
+stream<record{}, error> resultStream = jdbcClient->query(`SELECT NAME FROM STUDENT WHERE ID = ${<@untainted> studentId}`);
 ```
 
 Command-line arguments passed to Ballerina programs and inputs received through service resources are considered as tainted. Additionally, return values of certain functions are marked with the `@tainted` annotation to denote that the resulting value should be considered as untrusted data.
 
-For example, the `select` remote method of the `java:jdbc` client highlighted above returns a `@tainted table<record {}>|Error`. This means that any value read from a database is considered as untrusted.
+For example, the `query` remote method of the `java.jdbc` client highlighted above returns a `@tainted stream <record {}, sql:Error>`. This means that any value read from a database is considered as untrusted.
```

Comment:
```suggestion
For example, the `query` remote method of the `java.jdbc` client highlighted above returns an `@tainted stream <record {}, sql:Error>` error. This means that any value read from a database is considered as untrusted.
```

Reply:
`stream <record {}, sql:Error>` this is a stream type, this is not an error

Reply:
+1 to say "stream type" instead of "error".